### PR TITLE
Fix stackdriver integration tests

### PR DESCRIPTION
**Pull request by bochengchu**
**Please provide a description of this PR:**
https://github.com/istio/istio/pull/45467

### CODE REVIEW

**Review by lei-tang — APPROVED** [review]

**Comment by ericvn** [comment]
Thanks for the test updates.

File: tests/integration/telemetry/stackdriver/testdata/client_request_count.json.tmpl [hunk 1]
```diff
@@ -25,7 +25,8 @@
           "source_owner": "kubernetes://apis/apps/v1/namespaces/{{ .EchoNamespace }}/deployments/clt-{{ .ClusterName }}-v1",
           "source_principal": "spiffe://{{ .TrustDomain }}/ns/{{ .EchoNamespace }}/sa/default",
           "source_workload_name": "clt-{{ .ClusterName }}-v1",
-          "source_workload_namespace": "{{ .EchoNamespace }}"
+          "source_workload_namespace": "{{ .EchoNamespace }}",
+          "proxy_version": "unknown"
```
Comment (kyessenov) [inline tests/integration/telemetry/stackdriver/testdata/client_request_count.json.tmpl:29]: Do you know why it is unknown? Are we not sending ISTIO_VERSION anymore?

File: tests/integration/telemetry/stackdriver/testdata/client_tcp_connection_count.json.tmpl [hunk 2]
```diff
@@ -21,7 +21,8 @@
           "source_owner": "kubernetes://apis/apps/v1/namespaces/{{ .EchoNamespace }}/deployments/clt-{{ .ClusterName }}-v1",
           "source_principal": "spiffe://{{ .TrustDomain }}/ns/{{ .EchoNamespace }}/sa/default",
           "source_workload_name": "clt-{{ .ClusterName }}-v1",
-          "source_workload_namespace": "{{ .EchoNamespace }}"
+          "source_workload_namespace": "{{ .EchoNamespace }}",
+          "proxy_version": "{{ .IstioRevision }}"
```
Comment (howardjohn) [inline tests/integration/telemetry/stackdriver/testdata/client_tcp_connection_count.json.tmpl:25]: Why revision? revision != version

**Review by howardjohn — CHANGES_REQUESTED** [review]

File: tests/integration/telemetry/stackdriver/testdata/client_tcp_connection_count.json.tmpl (hunk 2 above)
Comment (bochengchu) [reply tests/integration/telemetry/stackdriver/testdata/client_tcp_connection_count.json.tmpl]: Fixed.

File: tests/integration/telemetry/stackdriver/testdata/client_request_count.json.tmpl (hunk 1 above)
Comment (bochengchu) [reply tests/integration/telemetry/stackdriver/testdata/client_request_count.json.tmpl]: Fixed.

File: tests/integration/telemetry/stackdriver/common.go [hunk 3]
```diff
@@ -275,6 +275,7 @@ func unmarshalFromTemplateFile(file string, out proto.Message, clName, trustDoma
 		"ClusterName":   clName,
 		"TrustDomain":   trustDomain,
 		"OnGCE":         metadata.OnGCE(),
+		"ProxyVersion":  os.Getenv("VERSION"),
```
Comment (howardjohn) [inline tests/integration/telemetry/stackdriver/common.go:278]: Not sure I understand this? Where is this set?

**Review by howardjohn — CHANGES_REQUESTED** [review]

File: tests/integration/telemetry/stackdriver/common.go (hunk 3 above)
Comment (howardjohn) [reply tests/integration/telemetry/stackdriver/common.go:278]: I think this is depending on it being run from makefiles which is not always how the tests are run. Can we avoid this implicit dependency? We have examples like `env.IstioSrc`.

Even still, I am not sure we should do this. The version in the repo we are running the test from is not neccesarily the version we deployed and are testing. IMO we should make this a fuzzy match

Comment (bochengchu) [reply tests/integration/telemetry/stackdriver/common.go]: Yes, that was from the makefiles.
I use `env.ReadVersion()` now and try to remove extra version information after pulling metrics from SD.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**